File: longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)
        visited = []
        visited.append(s[0])
        curr_max = 1
        i = 0
        j = 1
        while True:
            if s[j] in visited:
                visited.remove(s[i])
                i+=1
            else:
                visited.append(s[j])
                curr_max = max(curr_max, j-i+1)
                j+=1
            if j == len(s):
                return curr_max
        

File: medium/longest-substring-without-repeating-characters.cpp

#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        if (s.size() <= 1)
        {
            return s.size();
        }
        unordered_map<char, bool> m;
        int n = s.size();
        int i = 0;
        int j = 1;
        m[s[0]] = true;
        int curr_max = 0;
        while (true)
        {
            if (m.find(s[j]) == m.end())
            {
                m[s[j]] = true;
                curr_max = max(j - i + 1, curr_max);
                j++;
            }
            else
            {
                m.erase(s[i]);
                i++;
            }
            if (j == n)
            {
                break;
            }
        }
        return curr_max;
    }
};